blog/2022-08-05-paralus-audit-logs/index.md

```diff
@@ -0,0 +1,90 @@
+
+## Logging in Kubernetes
+
+Kubernetes as we all know is the most widely used container orchestration with a lot of moving parts. Nodes, clusters, pods, services, replica sets and a lot of other things.
+
+Logging activities across all these elements isn't an easy task, especially due to its ephemeral and dynamic nature.  With such a nature, Kubernetes provides three different types of logs:
+
+1. **Application logs:** Logs from apps deployed in clusters that help understand what's happening in the app.
+2. **Kubernetes Cluster components:** Logs from API-server, etcd, kube-proxy etc. that help troubleshoot cluster issues.
+3. **Kubernetes audit logs:** Logs related to API activity primarily used for audit & governance purposes.
+
+In this post we'll focus on Kubernetes Audit logs.
+
+Kubernetes provides a way to keep audit logs with a sequence of events that take place in a cluster. These are generated by users, applications or the control plane itself. Audit logs are generated for every request that generates an audit event and processed according to an audit policy.
+
+Audit Policy: An audit policy defines rules for all the events that should be recorded and the data that it will include.
+
+The audit levels in the policy are:
+
+* `none`: don't log any events
+* `metadata`: log only metadata
+* `request`: log metadata and request body
+* `requestresponse`: log metadata, request and response bodies.
+
+When an event is raised, the above rules are processed in order and are logged. We suggest you read more about [Kubernetes Audit Logging](https://kubernetes.io/docs/tasks/debug/debug-cluster/audit/) to know this better.
+
+Let us now look into how logging works in Paralus.
+
+## Audit Logs in Paralus
+
+All the API actions that are performed by any user using Paralus are recorded as part of the audit logs in Paralus. These provide a sequential log of activities that - *an audit trail*- that acts as evidence during compliance checks and other investigations.
+
+These logs are generated by actions performed by any authorized user across the organization. Further, any user across the organization can view the logs, but none of it could be deleted.
+
+Paralus allows you to track two types of logs:
+
+1. **System Logs**
+2. **Kubectl Logs**
+
+**System Logs:** These types of logs contain the entire history of activities performed in a specific organization. Activities like adding a user, creating a project, generating a new kubeconfig etc. are recorded.
+
+**Kubectl Logs:** These logs contain the history of kubectl commands executed from the web & API requests under each project. So all the kubectl commands executed either from the dashboard or pctl CLI tool will be recorded here.
```

Comment:
There are two kind of logs here:
1. Logs of kubectl commands (eg: kubectl get pods). This is only available when they are executing from web kubectl, not when they use their local kubectl.
2. API logs from relay. This is always available irrespective of where they execute the commands from.

Reply:
I'm confused, the UI says system and kubectl logs. Under kubectl logs I get the logs for all the commands executed and under System logs I'm able to see the logs for system actions like groups, roles etc. [Reference](https://www.paralus.io/docs/usage/audit-logs).

Reply:
Inside kubectl logs, there are two sub options, commands and api. The commands one is what comes from kube-prompt and api comes from relay. When looking at command logs, it might looks like you will be able to get the logs for command executed, but we are only able to give these when the user runs the command via web kubectl and not their local kubectl binary.

Reply:
Fixed.